eng/Versions.props

```diff
@@ -6,11 +6,6 @@
       This section defines executable versions of packages which are referenced via lower-version reference assemblies
       during the build.
     -->
-    <SystemCollectionsImmutableExecutableVersion>$(SystemCollectionsImmutableVersion)</SystemCollectionsImmutableExecutableVersion>
-    <SystemCollectionsImmutableExecutableVersion Condition="'$(SystemCollectionsImmutableExecutableVersion)' == ''">5.0.0</SystemCollectionsImmutableExecutableVersion>
-    <SystemReflectionMetadataExecutableVersion>$(SystemReflectionMetadataVersion)</SystemReflectionMetadataExecutableVersion>
-    <SystemReflectionMetadataExecutableVersion Condition="'$(SystemReflectionMetadataExecutableVersion)' == ''">5.0.0</SystemReflectionMetadataExecutableVersion>
-    <MicrosoftCodeAnalysisExecutableVersion>$(MicrosoftCodeAnalysisVersion)</MicrosoftCodeAnalysisExecutableVersion>
     <MicrosoftCodeAnalysisExecutableVersion Condition="'$(MicrosoftCodeAnalysisExecutableVersion)' == ''">3.8.0</MicrosoftCodeAnalysisExecutableVersion>
```

Comment:
As mentioned in the linked issue, for source-build we cannot utilize a ref pack for `Microsoft.CodeAnalysis` dependency versions.
Bumping even the minor version from `3.8.X` to `3.9.X` (or `3.7.X` to `3.8.X` which declared below) would result in build exceptions:

<details>
<summary>Exception</summary>

```
  /datadrive/source/roslyn-analyzers/artifacts/source-build/self/src/src/Roslyn.Diagnostics.Analyzers/Core/AbstractRunIterations`1.cs(39,32): error CS8600: Converting null literal or possible null value to non-nullable type. [/datadrive/source/roslyn-analyzers/artifacts/source-build/self/src/src/Roslyn.Diagnostics.Analyzers/Core/Roslyn.Diagnostics.Analyzers.csproj]
  /datadrive/source/roslyn-analyzers/artifacts/source-build/self/src/src/Roslyn.Diagnostics.Analyzers/Core/AbstractRunIterations`1.cs(40,18): error CS8604: Possible null reference argument for parameter 'method' in 'bool IMethodSymbolExtensions.IsBenchmarkOrXUnitTestMethod(IMethodSymbol method, ConcurrentDictionary<INamedTypeSymbol, bool> knownTestAttributes, INamedTypeSymbol? benchmarkAttribute, INamedTypeSymbol? xunitFactAttribute)'. [/datadrive/source/roslyn-analyzers/artifacts/source-build/self/src/src/Roslyn.Diagnostics.Analyzers/Core/Roslyn.Diagnostics.Analyzers.csproj]
  /datadrive/source/roslyn-analyzers/artifacts/source-build/self/src/src/Roslyn.Diagnostics.Analyzers/Core/AbstractApplyTraitToClass`1.cs(83,41): error CS8602: Dereference of a possibly null reference. [/datadrive/source/roslyn-analyzers/artifacts/source-build/self/src/src/Roslyn.Diagnostics.Analyzers/Core/Roslyn.Diagnostics.Analyzers.csproj]
  /datadrive/source/roslyn-analyzers/artifacts/source-build/self/src/src/Roslyn.Diagnostics.Analyzers/Core/AbstractApplyTraitToClass`1.cs(103,51): error CS8602: Dereference of a possibly null reference. [/datadrive/source/roslyn-analyzers/artifacts/source-build/self/src/src/Roslyn.Diagnostics.Analyzers/Core/Roslyn.Diagnostics.Analyzers.csproj]
```

</details>